Src/IronPython/Runtime/Binding/PythonProtocol.Operations.cs

```diff
@@ -1286,12 +1286,17 @@ internal static partial class PythonProtocol {
         }
 
         private static Expression/*!*/ MakeFallbackCompare(DynamicMetaObjectBinder/*!*/ binder, PythonOperationKind op, DynamicMetaObject[] types) {
-            return Ast.Call(
-                GetComparisonFallbackMethod(op),
-                PythonContext.GetCodeContext(binder),
-                AstUtils.Convert(types[0].Expression, typeof(object)),
-                AstUtils.Convert(types[1].Expression, typeof(object))
-            );
+            if (op == PythonOperationKind.Equal || op == PythonOperationKind.NotEqual ||
+                    op == PythonOperationKind.Compare) {
+                return Ast.Call(
+                    GetComparisonFallbackMethod(op),
+                    PythonContext.GetCodeContext(binder),
+                    AstUtils.Convert(types[0].Expression, typeof(object)),
+                    AstUtils.Convert(types[1].Expression, typeof(object))
+                );
+            }
+
+            return MakeBinaryThrow(binder, op, types).Expression;
```

Comment:
Any idea in what other cases `MakeBinaryThrow` is used? Just wondering if changing the error message to match CPython will break other error messages.

Reply:
MakeBinaryThrow is called for all binary operators I think; it looks like those also use the symbol in Python 3, so I just changed it here.

Reply:
Ok, looks like the error message is slightly different (CPython 3.8) for the comparison operators:
```
>>> 1 ^ None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for ^: 'int' and 'NoneType'
>>> 1 < None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```

Since the error messages basically mean the same thing I'm fine with diverging from CPython as long as there are no StdLib tests wanting to checking the actual message.